boards/arm/xiao_ble/xiao_ble.dts

```diff
@@ -139,30 +139,34 @@
 		compatible = "fixed-partitions";
 		#address-cells = <1>;
 		#size-cells = <1>;
-
-		/* The size of this partition ensures that MCUBoot can be built
-		 * with an RTT console, CDC ACM support, and w/o optimizations.
-		 */
-		boot_partition: partition@0 {
-			label = "mcuboot";
-			reg = <0x00000000 0x00012000>;
-		};
-
-		slot0_partition: partition@12000 {
-			label = "image-0";
-			reg = <0x00012000 0x00069000>;
+		
+		sd_partition: partition@0 {
+			label = "softdevice";
+			reg = <0x00000000 0x00027000>;
 		};
-		slot1_partition: partition@7b000 {
-			label = "image-1";
-			reg = <0x0007b000 0x00069000>;
+		
+		code_partition: partition@27000 {
+			label = "code_partition";
+			reg = <0x00027000 0x000c5000>;
 		};
-		scratch_partition: partition@e4000 {
-			label = "image-scratch";
-			reg = <0x000e4000 0x00018000>;
-		};
-		storage_partition: partition@fc000 {
+		        
+		/*
+		 * The flash starting at 0x000ec000 and ending at
+		 * 0x000f3fff is reserved for use by the application.
+		 */
+		        
+		/*
+		 * Storage partition will be used by FCB/LittleFS/NVS
+		 * if enabled.
+		 */
+		storage_partition: partition@ec000 {
 			label = "storage";
-			reg = <0x000fc000 0x00004000>;
+			reg = <0x000ec000 0x00008000>;
+		};
+		
+		boot_partition: partition@f4000 {
+			label = "adafruit_boot";
+			reg = <0x000f4000 0x0000c000>;
 		};
```

Comment:
I haven't quite parsed this aspect of the device tree, but can this support both the mcuboot way and the original bootloader?

Reply:
Not with the exact same board DTS, that I know of. Usually if you wanted to override something like that, you'd add a board specific overlay file to your *application* and then update the DTS w/ the correct partitions.

I don't believe Zephyr supports multiple partition "tables" in the DTS, that I've seen/been able to find.

Reply:
ah. i was hoping you could use chosen nodes for this to pick different default.

Reply:
There are certainly chosen nodes for the code region, slots, etc, but I couldn't find a chosen node to select the whole set of partitions.

Reply:
Yeah, there is no such mechanism yet. I think there was some thread on some PR to create something similar, but that never got upstreamed AFAIK.